cloudevents/extensions/bam.md

```diff
@@ -0,0 +1,129 @@
+"MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT",
+"RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as
+described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+However, the scope of these key words is limited to when this extension is
+used. For example, an attribute being marked as "REQUIRED" does not mean
+it needs to be in all CloudEvents, rather it needs to be included only when
+this extension is being used.
+
+## Attributes
+
+### bamtxid
+
+- Type: `String`
+- Description: A unique identifer for the instance of a transaction.
+  This identifier connects the actual processing in the distributed
+  system (e.g. payment, invoice, warehouse) with the model of this process.
+- Constraints
+  - REQUIRED
+  - MUST be a non-empty string
+  - RECOMMENDED as monotonically increasing and contiguous identifier
+    that is lexicographically-sortable.
+
+### bampid
+
+- Type: `String`
+- Description: A unique identifier for the model of the business process
+  that is associated with instance of the transaction `bamtxid`.
+  A business process is a collection of transactions. These transactions
+  can run in sequence or parallel (e.g. payment, invoice, warehouse).
+- Constraints
+  - REQUIRED
+  - MUST be a non-empty string
+  - RECOMMENDED a alphanumeric string that contains non-whitespace characters
+    and only hyphens, underscores, and periods.
+
+### bamptxid
+
+- Type: `String`
+- Description: A unique identifier for the model of a transaction that
+  constructs a business process (e.g. payment, invoice, warehouse).
+- Constraints
+  - REQUIRED
+  - MUST be a non-empty string
+  - RECOMMENDED a alphanumeric string that contains non-whitespace characters
+    and only hyphens, underscores, and periods.
+
+### bamptxsid
+
+- Type: `String`
+- Description: A unique identifier for the specific step in a business process
+  transaction (e.g. start, processing, finish).
+- Constraints
+  - REQUIRED
+  - MUST be a non-empty string
+  - RECOMMENDED a alphanumeric string that contains non-whitespace characters
+    and only hyphens, underscores, and periods.
+
+### bamptxsstatus
+
+- Type: `String`
+- Description: The status of the specific step in a business process
+  transaction (e.g. success, waiting, failure).
+- Constraints
+  - OPTIONAL
+  - if present, MUST be a non-empty string
+  - RECOMMENDED a alphanumeric string that contains non-whitespace characters
+    and only hyphens, underscores, and periods.
+
+### bamptxcompleted
+
+- Type: `Boolean`
+- Description: Indicates if the instance of the transaction (`bamtxid`) has
+  actually been completed, or if the transaction has somehow failed.
+  This is a mechanism to indicate a final completion or failure that is
+  not captures by the model of the business process..
+- Constraints
+  - OPTIONAL
+  - if present, MUST be a boolean value
+
+## Usage
+
+When this extension is used, producers MUST set the value of
```

Comment:
This should be changed to:
```
When this extension is used, producers MUST set the value of
the `bamtxid`, `bampid`,
`bamptxid`, and `bamptxsid`
attributes to the unique identifiers of the business process, transaction,
and transaction step associated with the event.

Intermediaries MUST NOT change the value of the `bamtxid`,
`bampid`, `bamptxid`,
and `bamptxsid` attributes.
```
since those other attribute names are no longer specified above. Again, I could see expanding the attribute name in parens for readability.